src/Common/src/TypeSystem/IL/Stubs/DelegateMarshallingMethodThunk.cs

```diff
@@ -93,6 +93,22 @@ public MetadataType DelegateType
             }
         }
 
+        private TypeDesc GetNativeMethodParameterType(TypeDesc managedType, MarshalAsDescriptor marshalAs, InteropStateManager interopStateManager, bool isReturn, bool isAnsi)
+        {
+            TypeDesc nativeType;
+            try
+            {
+                nativeType = MarshalHelpers.GetNativeMethodParameterType(managedType, marshalAs, interopStateManager, isReturn, isAnsi);
+            }
+            catch (NotSupportedException)
```

Comment:
We do this is couple of other places too. I will do the cleanup of this in a separate PR.